Mating/server.py

```diff
@@ -1,22 +1,44 @@
 import json
 from message import Message
-from network import UDP_IP, UDP_PORT
-from socket import socket, AF_INET, SOCK_DGRAM
+from network import UDP_PORT
+from socket import (socket, AF_INET, SOCK_DGRAM, SOL_SOCKET, SO_REUSEADDR,
+                    SO_REUSEPORT)
 from threading import Thread
 
 
 class Server(Thread):
-
-    def __init__(self, service=None):
+    """ Receives data from the Wi-Fi, by convention encoded as JSON strings,
+        and decides whether or not to respond.
+    """
+    def __init__(self, criterion=None, response=None):
+        """ Should override criterion and response functions in offsping. """
         Thread.__init__(self)
         self.s = socket(AF_INET, SOCK_DGRAM)
-        self.srv = service or default_service
+        self.address = ('', UDP_PORT)
+        self.s.setsockopt(SOL_SOCKET, SO_REUSEADDR, 1)
+        self.s.setsockopt(SOL_SOCKET, SO_REUSEPORT, 1)
+        self.s.bind(self.address)
+        # If criterion has not been set, use the default one
+        self.criterion = criterion or default_criterion
+        # If response has not been set, use the default one
+        self.response = response or default_response
 
     def run(self):
+        # Listen to the network
         while True:
-            data = self.srv()
-            self.s.sendto(json.dumps(data.__dict__), (UDP_IP, UDP_PORT))
+            packet, addr = self.s.recvfrom(1024)
```

Comment:
Yes, I also agree with Mateo.Whenever you can, you should use constant. It is alvays more explanatory then a real number. I looked also a rest of commits, it look just fine the way it was coded. So just add here some constant, and we can pull this commits.